### PR TITLE
#256 implemented storage and retrieval of visibility status

### PR DESCRIPTION
**Pull request by asaen**
#256 implemented storage and retrieval of visibility status of the document

### CODE REVIEW

**Comment by karato** [comment]
@darkled Many thanks, I will find someone to review it before we merge

**Comment by karato** [comment]
@carlosmiranda do this review

File: src/test/java/com/nerodesk/om/aws/AwsAttributesTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * @since 0.4
+ */
+public final class AwsAttributesTest {
+
+    /**
+     * AwsAttributes can be invisible by default.
+     * @throws IOException If fails.
+     */
+    @Test
+    public void isNotVisibleByDefault() throws IOException {
```
Comment (carlosmiranda) [inline src/test/java/com/nerodesk/om/aws/AwsAttributesTest.java]: @darkled Let's declare `Exception` being thrown here, as suggested [here](http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html?expand=1#test-method-names). And let's name the method `invisibleByDefault`.

File: src/test/java/com/nerodesk/om/aws/AwsAttributesTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        final AwsAttributes attr = new AwsAttributes();
+        MatcherAssert.assertThat(attr.visible(), Matchers.equalTo(false));
+    }
+
+    /**
+     * AwsAttributes can change visibility status properly.
+     * @throws IOException If fails.
+     */
+    @Test
+    public void changesVisibility() throws IOException {
```
Comment (carlosmiranda) [inline src/test/java/com/nerodesk/om/aws/AwsAttributesTest.java]: @darkled Here too, let's declare `Exception` being thrown, as suggested [here](http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html?expand=1#test-method-names).

File: src/test/java/com/nerodesk/om/aws/AwsAttributesTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ */
+public final class AwsAttributesTest {
+
+    /**
+     * AwsAttributes can be invisible by default.
+     * @throws IOException If fails.
+     */
+    @Test
+    public void isNotVisibleByDefault() throws IOException {
+        final AwsAttributes attr = new AwsAttributes();
```
Comment (carlosmiranda) [inline src/test/java/com/nerodesk/om/aws/AwsAttributesTest.java]: @darkled We can inline `new AwsAttributes()` in the next statement and avoid having this variable `attr`.

**Comment by carlosmiranda** [comment]
@darkled Some minor comments related to the test.

**Comment by asaen** [comment]
@carlosmiranda thanks, fixed, please take a look once more.

**Comment by carlosmiranda** [comment]
@rultor Good to merge

**Comment by yegor256** [comment]
@darkled hm... nice idea, but that's not what we need here :) we have to keep this status in persistent storage, not in memory

**Comment by carlosmiranda** [comment]
@darkled Any updates?

**Comment by asaen** [comment]
@carlosmiranda @yegor256 thank you for review, unfortunately for the moment I have no idea about this issue, closing PR.

**Comment by karato** [comment]
@carlosmiranda done, I added **18 mins** in payment `58075266`, 196 hours and 55 mins spent to complete this... review comments (c=3) added as a bonus... +18 added to your rating, current score is: [+4219](http://www.netbout.com/b/35023?open=rating)

**Pull request closed without merging**